gulp.config.js

```diff
@@ -81,12 +42,109 @@ module.exports = function() {
+        var themePaths = {
+            fonts: themeDir + '_fonts/**/*.{otf,svg,eot,ttf,woff,woff2}',
+            html: {
+                root: themeDir + '*.html',
+                nonRoot: [
+                    themeDir + '**/*.html',
+                    '!' + themeDir + email + '**/*',
+                    '!' + themeDir + '*.html',
+                ],
+            },
+            media: [
+                // Ignore media, declaring the _images folder directly moves all of its contents
+                // without the container folder getting in the way
+                themeDir + '_images/**/*.{png,gif,jpg,jpeg,ico,svg,mp4,ogv,webm,pdf}',
+                themeDir + '**/*.{png,gif,jpg,jpeg,ico,svg,mp4,ogv,webm,pdf}',
+                '!' + themeDir + '_fonts/*',
+            ],
+            misc: [
+                themeDir + '*.{htaccess,ico,xml}',
+                themeDir + 'humans.txt'
+            ],
+            robots: {
+                default: themeDir + 'robots.dev.txt',
+                prod: themeDir + 'robots.prod.txt',
+            },
+            styles: {
+                all: [
+                    themeDir + '**/*.scss',
+                    themeDir + '**/*.css',
+                ],
+            },
+            scripts: {
+                app: [
+                    temp + 'config.js', // this is a built file
+                    themeDir + '**/init.js',
+                    themeDir + '**/*.js',
+                    '!' + themeDir + '_lib/**/*', // don't clobber lib
+                ],
+                lib: [
+                    themeDir + '_lib/jquery.min.js',
+                    themeDir + '_lib/angular.min.js',
+                    themeDir + '_lib/**/*.min.js',
+                ],
+            },
+        };
+
+        mergeConfigPaths(config, themePaths);
+    }
+
+    // Adds to the config new paths
+    // iterates over newConfig path keys and adds paths to the current config
+    function mergeConfigPaths(currConfig, newConfig) {
+        for (var key in newConfig) {
+            var newConfigItem = newConfig[key];
+
+            // if there is not such a path key in the current config, create a new one
+            if (currConfig[key] == undefined) {
+                currConfig[key] = newConfigItem;
+
+            // if there is a single string in the current config,
+            // make an array of paths
+            } else if (typeof(currConfig[key]) === 'string') {
+                currConfig[key] = [currConfig[key], newConfigItem];
+
+            // if it's an array, add new paths to it
+            } else if (currConfig[key] instanceof Array) {
+                currConfig[key] = currConfig[key].concat(newConfigItem);
+
+            // if it's an object, recursively call mergeConfigPaths
+            } else if (typeof(currConfig[key]) === 'object') {
+                currConfig[key] = mergeConfigPaths(config[key], newConfigItem);
+            }
+        }
+
+        return currConfig;
+    }
```

Comment:
I think we need 4 functions here with names that clearly indicate what operation we are performing.  That will get rid of the comments on each test in the if-else statements and make the code much easier to follow.  It also follows our goal of having a function, **do one thing**.

It took me a while to digest what was going on here.

for example
```
if (currConfig[key] == undefined) {
     currConfig[key] = newConfigItem;
}
```

becomes
```
if ( isKeyUndefined(currConfig[key] ) {
     addKey(currConfig, key, newConfigItem)
}

isKeyUndefined(configKey) {
    return ((configKey == undefined) ? true : false);

addKey(config, newkey, newConfigPath) {
    .....
}
```